### PR TITLE
Remove IModelWriteRpc

### PR DESCRIPTION
**Pull request by kabentley**
Opening iModels for write access requires IPC. Remove previous attempt to allow writing through RPC.

Since Rpc no longer supports writing, `IModelRpcProps` no longer includes an `openMode` member.

Also, this PR introduces several breaking changes to APIs that use Changeset Ids. See `NextVersion.md` for details.

### CODE REVIEW

File: test-apps/ui-test-app/src/common/rpcs.ts [hunk 1]
```diff
@@ -3,32 +3,16 @@
...
 } from "@bentley/imodeljs-common";
 import { PresentationRpcInterface } from "@bentley/presentation-common";
-import { Config } from "@bentley/bentleyjs-core";
 
 /**
  * Returns a list of RPCs supported by this application
  */
 export function getSupportedRpcs(): RpcInterfaceDefinition[] {
 
-  if (Config.App.has("imjs_TESTAPP_ALLOW_WRITE") && (Config.App.get("imjs_TESTAPP_ALLOW_WRITE") === "1")) {
```
Comment (pmconne) [inline test-apps/ui-test-app/src/common/rpcs.ts:17]: Need to update README.md. @bbastings uses ui-test-app for testing editing tools. Does it still support editing standalone iModels?

File: core/frontend/src/CheckpointConnection.ts [hunk 2]
```diff
@@ -56,8 +51,8 @@ export class CheckpointConnection extends IModelConnection {
     const changeSetId = await IModelApp.hubAccess.getChangesetIdFromVersion({ requestContext, iModelId, version });
     requestContext.enter();
 
-    const iModelRpcProps: IModelRpcOpenProps = { contextId, iModelId, changeSetId, openMode };
-    const openResponse = await this.callOpen(requestContext, iModelRpcProps, openMode, routingContext);
+    const iModelRpcProps: IModelRpcOpenProps = { contextId, iModelId, changeSetId, openMode: OpenMode.Readonly };
```
[the remark this answers is not in the document]
Comment (bbastings) [reply core/frontend/src/CheckpointConnection.ts:54]: We need to replace the "model alignment" extension tools (I don't remember who wrote them? @ramanujam-raman). The editor package has improved replacements for the geolocation tools and modifying the project extents already.

File: test-apps/ui-test-app/src/common/rpcs.ts (hunk 1 above)
Comment (bbastings) [reply test-apps/ui-test-app/src/common/rpcs.ts:17]: Yes, ui-test-app lets you edit standalone imodels (needed to test toolsettings/tool assistance)

File: core/frontend/src/CheckpointConnection.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (calebmshafer) [reply core/frontend/src/CheckpointConnection.ts:54]: @bbastings, @nick4598 is going to work on model alignment. We should follow up internally on what work would need to be done to get it to the latest best practices.

**Review by pmconne — CHANGES_REQUESTED** [review]
Please change this line in ui-test-app's README.md: "Files from iModelHub can be edited."

**Review by bsteinbk — APPROVED** [review]

File: common/api/imodeljs-backend.api.md [hunk 3]
```diff
@@ -1258,10 +1256,10 @@ export interface DownloadJob {
 
 // @internal
 export interface DownloadRequest {
-    aliasFiles?: string[];
-    checkpoint: CheckpointProps;
+    readonly aliasFiles?: string[];
```
Comment (pmconne) [inline common/api/imodeljs-backend.api.md:1259]: Should `aliasFiles` be a `ReadonlyArray<string>`?

File: core/frontend/src/CheckpointConnection.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (kabentley) [reply core/frontend/src/CheckpointConnection.ts]: I have now removed `openMode` from `IModelRpcProps`. I also changed (non-backwards-compatibly) the changeSetId/changeSetIndex members of that interface.

File: common/api/imodeljs-backend.api.md (hunk 3 above)
Comment (kabentley) [reply common/api/imodeljs-backend.api.md:1259]: probably. `readonly aliasFiles?: ReadonlyArray<string>`

Comment (kabentley) [reply common/api/imodeljs-backend.api.md]: done

**Comment by kabentley** [comment]
> Please change this line in ui-test-app's README.md: "Files from iModelHub can be edited."

I updated the READEME (although i don't know how to test that it works.)

**Comment by pmconne** [comment]
> I updated the READEME (although i don't know how to test that it works.)

As far as I can tell, editing iModels from the hub hasn't worked for a long time - [example](https://github.com/imodeljs/imodeljs/discussions/2036#discussioncomment-1163489).
README should say only editable standalone iModels are supported for editing. Of course most people reading that - such as user who asked the linked question - have no idea how to make an editable standalone iModel.

**Comment by kabentley** [comment]
> As far as I can tell, editing iModels from the hub hasn't worked for a long time - [example](https://github.com/imodeljs/imodeljs/discussions/2036#discussioncomment-1163489).
> README should say only editable standalone iModels are supported for editing. Of course most people reading that - such as user who asked the linked question - have no idea how to make an editable standalone iModel.

@bsteinbk The whole "local file stage" support should be reworked. This test app *always* works on local files, since the backend is always on the same computer as the frontend. When you download a briefcase (with briefcaseId !== 0), you put it somewhere on your disk. You should be able to edit them using `BriefcaseConnection`, as you already do in `LocalFileStage.ts`. The concept of editing a standalone iModel should be removed - nobody needs that.

**Review by pmconne — APPROVED** [review]

**Pull request merged**